flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/FlinkOperator.java

```diff
@@ -78,12 +88,27 @@ public static void main(String... args) {
                         jobReconciler,
                         sessionReconciler);
 
-        FlinkControllerConfig controllerConfig = new FlinkControllerConfig(controller);
+        controllerConfig = new FlinkControllerConfig(controller);
         controller.setControllerConfig(controllerConfig);
         controllerConfig.setConfigurationService(configurationService);
+    }
 
+    public void run() {
         operator.register(controller, controllerConfig);
         operator.installShutdownHook();
         operator.start();
     }
+
+    protected ConfigurationService getConfigurationService() {
+        return new ConfigurationServiceDecorator(DefaultConfigurationService.instance()) {
+            @Override
+            public ExecutorService getExecutorService() {
+                return Executors.newCachedThreadPool();
```

Comment:
Should this always be unlimited size? Maybe we should make this configurable, what do you think

Reply:
The only case I can see when I would set a limit is when I would be getting out of memory. In such a case by setting a limit we could keep the operator running.

I've added a new config option for that: `RECONCILIATION_MAX_PARALLELISM`